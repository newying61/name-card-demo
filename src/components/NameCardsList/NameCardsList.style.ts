import styled from 'styled-components';

export const Wrapper = styled.div`
  display: flex;
  flex-wrap: wrap;
  justify-content: center;

  @media only screen and (min-width: 600px) {
    justify-content: flex-start;
  }
`;
