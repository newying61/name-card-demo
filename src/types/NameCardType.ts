export interface NameCard {
  id: string;
  firstName: string;
  lastName: string;
  dateOfBirth: string;
}
