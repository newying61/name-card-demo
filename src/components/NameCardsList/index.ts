import NameCardsList from './NameCardsList';

export default NameCardsList;
