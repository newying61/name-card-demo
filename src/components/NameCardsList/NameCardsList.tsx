import React from 'react';
import { NameCard as NameCardType } from '../../types/NameCardType';
import { Wrapper } from './NameCardsList.style';
import NameCard from '../NameCard';

interface NameCardsListProps {
  nameCards: NameCardType[];
  onEdit: (card: NameCardType) => void;
  onDelete: (card: NameCardType) => void;
}

const NameCardsList: React.FC<NameCardsListProps> = ({ nameCards, onEdit, onDelete }) => {
  return (
    <Wrapper>
      {nameCards.map((card) => {
        return (
          <NameCard
            key={card.id}
            card={card}
            onEdit={onEdit}
            onDelete={onDelete}
          />
        );
      })}
    </Wrapper>
  );
}

export default NameCardsList;