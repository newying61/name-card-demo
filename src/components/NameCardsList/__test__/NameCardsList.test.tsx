import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { NameCard } from '../../../types/NameCardType';
import NameCardsList from '../NameCardsList';

afterEach(cleanup);

describe('NameCardsList', () => {
    it('should render without crash when nameCards list is null', () => {
      const { container } = render(
        <NameCardsList nameCards={null} onEdit={jest.fn()} onDelete={jest.fn()}  />
      );

      expect(container.firstChild).toMatchSnapshot();
  });

    it('should render without crash when nameCards list is empty', () => {
      const { container } = render(
        <NameCardsList nameCards={[]} onEdit={jest.fn()} onDelete={jest.fn()}  />
      );

      expect(container.firstChild).toMatchSnapshot();
  });

  it('should render without crash when nameCards', () => {
    const cards: NameCard[] = [{
      id: 'test',
      firstName: 'test',
      lastName: 'test',
      dateOfBirth: '2019/01/01'
    }];
    const { container } = render(
      <NameCardsList nameCards={cards} onEdit={jest.fn()} onDelete={jest.fn()}  />
    );

    expect(container.firstChild).toMatchSnapshot();
});
});