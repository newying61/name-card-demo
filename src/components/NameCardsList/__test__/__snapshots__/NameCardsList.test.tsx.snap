// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NameCardsList should render without crash when nameCards 1`] = `
<div
  class="sc-bdVaJa eYASFs"
>
  <div
    class="sc-ifAKCX plnT"
  >
    <ul
      class="list-group list-group-flush"
    >
      <li
        class="list-group-item"
      >
        <dt>
          First name
        </dt>
        <dd
          class="sc-bwzfXH eIjNIe"
        >
          test
        </dd>
      </li>
      <li
        class="list-group-item"
      >
        <dt>
          Last name
        </dt>
        <dd
          class="sc-bwzfXH eIjNIe"
        >
          test
        </dd>
      </li>
      <li
        class="list-group-item"
      >
        <dt>
          Date of birrth
        </dt>
        <dd
          class="sc-bwzfXH eIjNIe"
        >
          2019/01/01
        </dd>
      </li>
    </ul>
    <div
      class="sc-bxivhb jcdCai"
    >
      <button
        class="sc-htpNat eZpMOu btn btn-primary"
        type="button"
      >
        Edit
      </button>
      <button
        class="sc-htpNat eZpMOu btn btn-danger"
        type="button"
      >
        Delete
      </button>
    </div>
  </div>
</div>
`;

exports[`NameCardsList should render without crash when nameCards list is empty 1`] = `
<div>
  No cards
</div>
`;

exports[`NameCardsList should render without crash when nameCards list is null 1`] = `
<div>
  No cards
</div>
`;
